src/people.rs

```diff
@@ -361,10 +462,25 @@ impl ContextPeopleExt for Context {
             .map_or(0, |data_container| data_container.current_population)
     }
 
-    fn add_person(&mut self) -> PersonId {
-        let person_id = self.get_data_container_mut(PeoplePlugin).add_person();
+    fn add_person<T: InitializationList>(&mut self, props: T) -> Result<PersonId, IxaError> {
+        let data_container = self.get_data_container_mut(PeoplePlugin);
+        // Verify that every property that was supposed to be provided
+        // actually was.
+        data_container.check_initialization_list(&props)?;
+
+        // Actually add the person. Nothing can fail after this point because
+        // it would leave the person in an inconsistent state.
+        let person_id = data_container.add_person();
+
+        // Initialize the properties. We set |is_initializing| to prevent
+        // set_person_property() from generating an event.
+        data_container.is_initializing = true;
```

Comment:
Wdyt about using a with_initializing method that takes a closure or something like that?

Reply:
I don't think I understand what you are proposing here.